--- lib/sharedbook.c	(revision 16072)
+++ lib/sharedbook.c	(revision 16073)
@@ -125,12 +125,18 @@
       if(sparsecount==0)count++;
   }
   
-  /* sanity check the huffman tree; an underpopulated tree must be rejected. */
-  for(i=1;i<33;i++)
-    if(marker[i] & (0xffffffffUL>>(32-i))){
-      _ogg_free(r);
-      return(NULL);
-    }
+  /* sanity check the huffman tree; an underpopulated tree must be
+     rejected. The only exception is the one-node pseudo-nil tree,
+     which appears to be underpopulated because the tree doesn't
+     really exist; there's only one possible 'codeword' or zero bits,
+     but the above tree-gen code doesn't mark that. */
+  if(sparsecount != 1){
+    for(i=1;i<33;i++)
+      if(marker[i] & (0xffffffffUL>>(32-i))){
+	_ogg_free(r);
+	return(NULL);
+      }
+  }
 
   /* bitreverse the words because our bitwise packer/unpacker is LSb
      endian */
